Build problems on SmartOS (illumos distro)

Node newb here. Sorry for not knowing much, but I am unable to get this to install on recent SmartOS. I will be happy to provide any information necessary to debug.

Node version: 4.1.2
npm version: 2.14.4

Here is the compile error:

```
npm install -g modern-syslog
-
> modern-syslog@1.1.1 install /opt/local/lib/node_modules/modern-syslog
> node-gyp rebuild

make: Entering directory '/opt/local/lib/node_modules/modern-syslog/build'
  CXX(target) Release/obj.target/core/core.o
../core.cc: In function 'void {anonymous}::Init(Nan::ADDON_REGISTER_FUNCTION_ARGS_TYPE)':
../core.cc:131:5: error: 'LOG_PERROR' was not declared in this scope
core.target.mk:92: recipe for target 'Release/obj.target/core/core.o' failed
make: *** [Release/obj.target/core/core.o] Error 1
make: Leaving directory '/opt/local/lib/node_modules/modern-syslog/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/opt/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System SunOS 5.11
gyp ERR! command "/opt/local/bin/node" "/opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /opt/local/lib/node_modules/modern-syslog
gyp ERR! node -v v4.1.2
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm ERR! SunOS 5.11
npm ERR! argv "/opt/local/bin/node" "/opt/local/bin/npm" "install" "-g" "modern-syslog"
npm ERR! node v4.1.2
npm ERR! npm  v2.14.4
npm ERR! code ELIFECYCLE

--- a/README.md
+++ b/README.md
@@ -134,7 +134,7 @@ Levels are listed from highest priority, to lowest:
 Object of properties:
 
 - `LOG_CONS`: Log to console if there is error logging to syslog.
-- `LOG_PERROR`: Log to stderr as well as syslog.
+- `LOG_PERROR`: Log to stderr as well as syslog. (no-op on Solaris)
 - `LOG_PID`: Log process' PID with each message.
 
 These are unlikely to be useful, but are provided for completeness:

--- a/core.cc
+++ b/core.cc
@@ -128,6 +128,10 @@ NAN_MODULE_INIT(Init) {
     DEFINE(LOG_CONS);
     DEFINE(LOG_NDELAY);
     DEFINE(LOG_ODELAY);
+#ifndef LOG_PERROR
+// not defined on Solaris but we want the exported object to be consistent
+#define LOG_PERROR 0x0 // no-op
+#endif
     DEFINE(LOG_PERROR);
     DEFINE(LOG_PID);
     DEFINE(LOG_NOWAIT);

